packages/neuron-ui/src/components/SUDTCreateDialog/index.tsx

```diff
@@ -153,10 +153,24 @@ const SUDTCreateDialog = ({
         value: payload,
         dataset: { field: type },
       } = e.target
-      const tokenInfo = tokenInfoList.find(ti => payload === ti.tokenID)
-      if (type === 'tokenId' && tokenInfo) {
-        const { tokenID, ...rest } = tokenInfo
-        dispatch({ type: 'import', payload: { ...rest, tokenId: tokenID } })
+      if (
+        type === 'tokenId' &&
+        payload.startsWith('0x') &&
+        payload.length === TOKEN_ID_LENGTH &&
+        !Number.isNaN(+payload)
+      ) {
+        const tokenInfo = tokenInfoList.find(ti => payload === ti.tokenID)
+        if (tokenInfo) {
+          const { tokenID, ...rest } = tokenInfo
+          dispatch({ type: 'import', payload: { ...rest, tokenId: tokenID } })
+        } else {
+          getSUDTTokenInfo({ tokenId: payload }).then(res => {
+            if (isSuccessResponse(res) && res.result) {
+              dispatch({ type: 'import', payload: { ...res.result } })
+            }
+          })
+          dispatch({ type, payload })
+        }
```

Comment:
This looks quite complicated. Can we find a way to simplify these condition checks a bit?